packages/proxy/lib/http/index.ts

```diff
@@ -182,7 +183,13 @@ export function _runStage (type: HttpStages, ctx: any, onError: Function) {
       const fullCtx = {
         next: () => {
           fullCtx.next = () => {
-            throw new Error('Error running proxy middleware: Cannot call this.next() more than once in the same middleware function. Doing so can cause unintended issues.')
+            const error = new Error('Error running proxy middleware: Detected `this.next()` was called more than once in the same middleware function. This can cause unintended issues and may indicate an error in your middleware logic or configuration.')
+
+            if (ctx.error) {
+              error.message = error.message += `\nThis middleware invocation previously encountered an error which may be related: ${ctx.error}`
+            }
```

Comment:
Maybe use `error.cause` here so the other error stack trace is also printed? https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause

```suggestion
            if (ctx.error) {
              error.message = error.message += '\nThis middleware invocation previously encountered an error which may be related, see `error.cause`'
              error.cause = ctx.error
            }
```